src/libsync/pushnotifications.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PUSHNOTIFICATIONS_H
+#define PUSHNOTIFICATIONS_H
+
+#include "websocket.h"
+
+namespace OCC {
+
+class Account;
+class AbstractCredentials;
+
+class PushNotifications : public QObject
+{
+    Q_OBJECT
+
+public:
+    // Question: Is it okay to pass a raw account pointer? Because PushNotifications don't really own the account. The PushNotifications
+    // object is owned by the account state. Why not passing AccountState? Maybe would be better but this makes testing more difficult.
+    PushNotifications(Account *account, QSharedPointer<AbstractWebSocket> webSocket = QSharedPointer<AbstractWebSocket>(new WebSocket));
+    void reconnect();
+    bool filesPushNotificationsAvailable() const;
+
+signals:
+    void filesChanged(Account *account);
```

Comment:
Since we're there add the signals for the other notification types, you won't make use of it in that PR but that'll be less work to do in the next PR and doesn't add much here.

Reply:
Still stand by this.

Reply:
What are there for other notification types? Is there any specification or source code from which I can get that?